Add documentation for `#[MapUploadedFile]` attribute (#19807)
symfony/symfony-docs: controller.rst

    The ``type`` option of ``#[MapRequestPayload]`` was introduced in Symfony 7.1.

Mapping Uploaded File
~~~~~~~~~~~~~~~~~~~~~~~

> You can map ``Uploaded Files`` to the controller arguments and add ``constraints``.

Reviewer: ```suggestion
You can map ``UploadedFile``s to the controller arguments and add ``constraints``.
```

By default the argument resolver will try to match the argument name with the submitted file name.

> The constraints defined by you are performed before injecting the uploaded file into the controller argument.

Reviewer: ```suggestion
The constraints defined by you are performed before injecting the `UploadedFile` into the controller argument.
```

An ``Unprocessable Content HTTP``` exception is thrown if a constraint violation is detected.

>     #[Route('/user/picture', methods: ['PUT'])]

Reviewer: Please add the use statements of the UploadedFile, the Assert and the MapUploadedFile

    final class UserPictureController
    {
        public function _invoke(
            #[MapUploadedFile( new Assert\File(mimeTypes: ['image/png', 'image/jpeg']) )]
            ?UploadedFile $picture
        ): Response {
            // ...
        }
    }

> You can also map a collection of uploaded files to array or variadic arguments::

Reviewer: ```suggestion
You can also map a collection of `UploadedFile`s::
```

>     #[MapUploadedFile( new Assert\File(mimeTypes: ['application/pdf']) )]

Reviewer: ```suggestion
    #[MapUploadedFile(new Assert\File(mimeTypes: ['application/pdf']))]
```

    array $documents

>     #[MapUploadedFile( new Assert\File(mimeTypes: ['application/pdf']) )]

Reviewer: ```suggestion
    #[MapUploadedFile(new Assert\File(mimeTypes: ['application/pdf']))]
```

    UploadedFile ...$documents

Further, you can match the argument with a custom name,

> and also change the ``HTTP Status Code`` thrown when a constraint violation are detected.

Reviewer: ```suggestion
and also change the thrown HTTP status code, when constraint violations are detected::
```

    #[MapUploadedFile(
        constraints: new Assert\File(mimeTypes: ['application/pdf']),
        name: 'something-else',
        validationFailedStatusCode: Response::HTTP_BAD_REQUEST
    )]
    UploadedFile $document

> 

Reviewer: please add a `versionadded` directive
```suggestion

.. versionadded:: 7.1

    The `#[MapUploadedFile]` attribute was introduced in Symfony 7.1.

```

Mapping Uploaded File
~~~~~~~~~~~~~~~~~~~~~

You can map ``UploadedFile``s to the controller arguments and optionally bind ``Constraints`` to them.
The argument resolver fetches the ``UploadedFile`` based on the argument name.

.. code-block:: php-attributes

    namespace App\Controller;

    use Symfony\Component\HttpFoundation\File\UploadedFile;
    use Symfony\Component\HttpFoundation\Response;
    use Symfony\Component\HttpKernel\Attribute\MapUploadedFile;
    use Symfony\Component\Routing\Attribute\Route;
    use Symfony\Component\Validator\Constraints as Assert;

    #[Route('/user/picture', methods: ['PUT'])]
    class UserPictureController
    {
        public function _invoke(
            #[MapUploadedFile(new Assert\File(mimeTypes: ['image/png', 'image/jpeg']))]
            UploadedFile $picture
        ): Response {
            // ...
        }
    }

.. tip::

    The bound ``Constraints`` are performed before injecting the ``UploadedFile`` into the controller argument.
    When a constraint violation is detected an ``HTTPException`` is thrown and the controller's
    action is not executed.

Mapping ``UploadedFile``s with no custom settings.

> .. code-block:: php-attributes

Reviewer 2: Are these directives really needed, as there is only one code block and not tabs?

Author: I had to explicitly set `php-attributes` markup format to these code blocks to pass the checkers.
